infobip-typescript-generator-extension-api/src/main/java/com/infobip/typescript/transformer/ClassTransformerDecoratorExtension.java

```diff
@@ -114,6 +140,18 @@ private TsPropertyModel getDecorators(SymbolTable symbolTable, TsBeanModel model
                        .collect(Collectors.toList());
     }
 
+    private boolean shouldNotBeDecorated(Class<?> type) {
+        return isBuiltIn(type) || isTsTypeResolutionUnsupported(type);
+    }
+
+    private boolean isTsTypeResolutionUnsupported(Class<?> type) {
+        return type.isEnum() || PresentPropertyJsonHierarchy.class.isAssignableFrom(type);
+    }
+
+    private boolean isBuiltIn(Class<?> type) {
+        return Optional.ofNullable(type.getPackage()).map(Package::getName).orElse("").startsWith("java");
```

Comment:
Question:
We are skipping decoration for built in java types, if so i would rename method to isBuiltInType

Reply:
Done